package-builder/flox-build.mk

```diff
@@ -414,24 +446,30 @@ define BUILD_template =
   .PHONY: build/$(_pname)
   build/$(_pname): $(_result)
 
-  # Accumulate a list of known build targets for the "build" target.
-  build_targets += $(_result)
 endef
 
-# Glean the sandbox mode from manifest metadata.
+# Glean various values from locked manifest as we call the template.
 $(foreach build,$(BUILDS), \
   $(eval _pname = $(notdir $(build))) \
   $(eval _sandbox = $(shell \
     $(_jq) -r '.manifest.build."$(_pname)".sandbox' $(MANIFEST_LOCK))) \
   $(eval _version = $(shell \
-    $(_jq) -r '.manifest.build."$(_pname)".version' $(MANIFEST_LOCK))) \
+    $(_jq) -r '.manifest.build."$(_pname)".version // "unknown"' $(MANIFEST_LOCK))) \
   $(if $(filter null off,$(_sandbox)), \
     $(eval $(call BUILD_template,local)), \
     $(eval $(call BUILD_template,nix_sandbox))))
 
-# Finally, we create the "build" target to build all known packages.
+# Combine JSON build data for each build and write to BUILD_RESULT_FILE.
+# Mark it as phony to force it to be evaluated every time.
+.PHONY: $(BUILD_RESULT_FILE)
+$(BUILD_RESULT_FILE): $(foreach pname,$(PACKAGES),$($(subst -,_,$(pname))_buildMetaJSON))
+	$(_VV_) [ -n "$^" ] || ( echo "ERROR: PACKAGES not defined or empty" 1>&2; exit 1 )
+	$(_VV_) $(_jq) -s . $^ > $@
```

Comment:
(reviewer commentary) Filing `"$^"` under new and difficult-to-Google ways for Make to surprise me 😆

Reply:
$@, $< and $^ are literally the first variables to learn with make. Gnu make adds $(@D) and $(@F) which are also super useful. Those 5 variables will take you far!